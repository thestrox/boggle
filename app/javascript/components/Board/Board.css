.board {
    background-color: lightblue;
}

.board-row {
    display: flex;
    flex-direction: row;
}