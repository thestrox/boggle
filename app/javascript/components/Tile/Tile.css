.tile-container {
    background-color: lightcyan;
    width: 50px;
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 10px;
    cursor: pointer;
}

.letter-span {
    font-family: Roboto;
    font-size: 16px;
}
