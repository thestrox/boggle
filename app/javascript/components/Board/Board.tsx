import React from "react";
import Tile from "../Tile/Tile";
import "./Board.css"

export type BoardProps = {
    board: string[][],
    className: string,
    onTileClick: (row: number, col: number) => void
}

export default function Board({ board, className, onTileClick }: BoardProps) {

    return (
        <div className={`board ${className}`}>
            {board.map((row, rowIndex) => {
                return (
                    <div className="board-row" key={rowIndex}>
                        {row.map((letter, colIndex) => {
                            return (<Tile key={colIndex} onClick={onTileClick.bind(this, rowIndex, colIndex)} letter={letter} />);
                        })}
                    </div>
                );
            })}
        </div >
    );

}