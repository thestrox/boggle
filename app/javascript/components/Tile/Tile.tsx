import React from "react";
import "./Tile.css"

export type TileProp = {
    letter: string,
    onClick: () => void
}

export default function Tile({ letter, onClick }: TileProp) {

    return (
        <div onClick={onClick} className="tile-container">
            <span className="letter-span">{letter}</span>
        </div>
    );
}