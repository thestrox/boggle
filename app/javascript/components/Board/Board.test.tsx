import Board from './Board';
import React from 'react';
import { shallow } from 'enzyme';
import Tile from '../Tile/Tile';

function setup() {
    const props = {
        board: [
            ["a", "b", "c", "d"],
            ["e", "f", "g", "h"],
            ["i", "j", "k", "l"],
            ["m", "n", "o", "p"]
        ],
        className: "test-class",
        onTileClick: jest.fn()
    };

    const component = shallow(<Board {...props}/>);
    return {
        props,
        component
    }
}

describe('Board Component', () => {
    it('should render correctly with given props', () => {
        const {component, props} = setup();

        component.find('div').forEach((node, index) => {
            if(index === 0) {
                expect(node.hasClass('board')).toBe(true);
                expect(node.hasClass(props.className)).toBe(true);
            } else {
                expect(node.hasClass('board-row')).toBe(true);
            }
        });
        
        let letters: string[] = [].concat(...props.board);
        component.find(Tile).forEach((node, index) => {
            expect(node.prop('letter')).toEqual(letters[index]);
        });
    });

    it('should call onTileClick', () => {
        const {component, props} = setup();
        component.find(Tile).forEach(node => {
            node.props().onClick();
        });
        expect(props.onTileClick.mock.calls.length).toBe(16); 
    });

    it('should test snapshot renders', () => {
        const {component} = setup();
        expect(component).toMatchSnapshot();
    });
})