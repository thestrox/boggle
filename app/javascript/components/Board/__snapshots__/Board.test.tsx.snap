// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component should test snapshot renders 1`] = `
<div
  className="board test-class"
>
  <div
    className="board-row"
    key="0"
  >
    <Tile
      key="0"
      letter="a"
      onClick={[Function]}
    />
    <Tile
      key="1"
      letter="b"
      onClick={[Function]}
    />
    <Tile
      key="2"
      letter="c"
      onClick={[Function]}
    />
    <Tile
      key="3"
      letter="d"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
    key="1"
  >
    <Tile
      key="0"
      letter="e"
      onClick={[Function]}
    />
    <Tile
      key="1"
      letter="f"
      onClick={[Function]}
    />
    <Tile
      key="2"
      letter="g"
      onClick={[Function]}
    />
    <Tile
      key="3"
      letter="h"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
    key="2"
  >
    <Tile
      key="0"
      letter="i"
      onClick={[Function]}
    />
    <Tile
      key="1"
      letter="j"
      onClick={[Function]}
    />
    <Tile
      key="2"
      letter="k"
      onClick={[Function]}
    />
    <Tile
      key="3"
      letter="l"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
    key="3"
  >
    <Tile
      key="0"
      letter="m"
      onClick={[Function]}
    />
    <Tile
      key="1"
      letter="n"
      onClick={[Function]}
    />
    <Tile
      key="2"
      letter="o"
      onClick={[Function]}
    />
    <Tile
      key="3"
      letter="p"
      onClick={[Function]}
    />
  </div>
</div>
`;
